src/sentry/integrations/discord/integration.py

```diff
@@ -46,8 +49,40 @@ class DiscordIntegrationProvider(IntegrationProvider):
     features = frozenset([IntegrationFeatures.CHAT_UNFURL])
     requires_feature_flag = True  # remove this when we remove the discord feature flag
 
+    # https://discord.com/developers/docs/topics/oauth2#shared-resources-oauth2-scopes
+    oauth_scopes = frozenset(["applications.commands", "bot"])
+
+    # Visit the bot tab of your app in the Discord developer portal for a tool to generate this
+    bot_permissions = 2048
+
+    setup_dialog_config = {"width": 600, "height": 900}
+
     def get_pipeline_views(self) -> Sequence[PipelineView]:
-        return super().get_pipeline_views()
+        return [DiscordInstallPipeline(self.get_bot_install_url())]
+
+    def build_integration(self, state: Mapping[str, object]) -> Mapping[str, object]:
+        guild_id = str(state.get("guild_id"))
+        guild_name = DiscordClient()._get_guild_name(guild_id)
+        return {
+            "name": guild_name,
+            "external_id": guild_id,
+        }
```

Comment:
Rather than call the `_get_guild_name` method which was annotated to only be used here, you can just get rid of it entirely and do the request here. That way future devs using `DiscordClient` don't end up using methods incorrectly.

```python
DiscordClient().get(f'/guilds/{guild_id}', headers={"Authorization": f"Bot {self.bot_token}"})
```

Reply:
Good call